services/src/main/java/org/keycloak/protocol/oidc/endpoints/LogoutEndpoint.java

```diff
@@ -212,6 +212,9 @@ public Response logoutToken() {
         try {
             session.clientPolicy().triggerOnEvent(new LogoutRequestContext(form));
         } catch (ClientPolicyException cpe) {
+            if (MtlsHoKTokenUtil.CERT_VERIFY_ERROR_DESC.equals(cpe.getErrorDetail())) {
```

Comment:
IMO it will be ideal, if there is no need to have the code here in the class LogoutEndpoint and manually "translate" the ClientPolicyException into proper ErrorResponse this way.

Because if someone creates custom executor, which want to return special error/errorDescription/responseStatus, he won't be able to do that without changing all the catch blocks in the endpoints. And implementors of the executor providers won't have an option to change the endpoints...

Is it possible that eiter:
1) ClientPolicyException will encapsulate all the informations needed to construct the proper ErrorResponse. For example the "status" code can be added as an additional field to ClientPolicyException (it will be BAD_REQUEST by default and "error" field would be INVALID_REQUEST by default)

2) There will be another method on the ClientPolicyExecutor to translate ClientPolicyException into the proper JAX-RS exception

It seems to me that (1) is simpler to do and probably ok to achieve this task? WDYT?

Reply:
Yes, you are right. I would like to choose option 1.